Fix-User-Guide.md

```diff
@@ -0,0 +1,252 @@
+- - or it can be separated in a external file, that is called in the FLUX-Process as in the code-snipped above
+- when using it in a JAVA-Process, just add the library to your process
+
+## Record-based and metadata manipulating approach
+While Metafature processes the data as a stream, the fix module does not it buffers the incoming stream to distinct records.
+So that you can manipulate all metadata-elements of a record at once and without focussing about the order of the incoming stream - which was a really big hassle in the stream-based MORPH.
+The incoming record then can be manipulated, fields can be changed, removed or added. This also differs from the approach in MORPH where you construct a new record and a new data stream, whereas you change stuff in the record in FIX and "only" change the data stream in Metafacture.
+
+
+## Basic concepts
+The four main concepts of FIX (introduced by catmandu) are [functions](https://librecat.org/Catmandu/#functions), [selector](https://librecat.org/Catmandu/#selectors), [conditionals](https://librecat.org/Catmandu/#conditionals) and [binds](https://librecat.org/Catmandu/#binds). The following code snippet shows examples of eachs of these concepts:
+
+
+```PERL
+
+# Simple fix function
+
+add_field("hello", "world")
+remove_field("my.deep.nested.junk")
+copy_field("stats", "output.$append")
+
+# Conditionals
+
+if exists("error")
+  set_field("is_valid", "no")
+  log("error")
+elsif exists("warning")
+  set_field("is_valid", "yes")
+  log("warning")
+else
+  set_field("is_valid", "yes")
+end
+
+# Binds - Loops
+
+do list(path: "foo", "var": "$i")
+  add_field("$i.bar", "baz")
+end
+
+# Selector
+if exists("error")
+   reject()
+end
+
+```
+
+**Functions** are used to add, change, remove or otherwise manipulate elements.
+
+**Conditionals** are used to control the processing of function so that they are not process with every workflow but only under certain conditions.
+
+**Selectors** can be used as hghlevel filter to filter the records you want.
+
+**Binds** are wrappers for one or more fixes. They give extra control functionality for fixes such as loops.
+All binds have the same syntax:
+
+```PERL
+do Bind(params,…)
+   fix(..)
+   fix(..)
+end
+```
+
+Find here a [list of all function, selectors, binds and conditionals](/Fix-function-and-Cookbook.md).
+
+
+## Addressing Pieces of Data - of: FIX-Path and the record structure in FIX
+
+Internally FIX knows arrays, objects/hashes and simple elements. How a format is translated is depending on the `decode-...` command in the MF Workflow. Only one thing is specific to the fix, as in Catmandu a repeated field is translated into a list depending on the real input data of the single record and elements with the suffix `[]` are interpreted as arrays.
+
+Since function manipulate, add or remove elements in a record, it is essential to understand the way on can adress source or target elements.
```

Comment:
```suggestion
Since functions manipulate, add or remove elements in a record, it is essential to understand the way you can address source or target elements.
```